packages/@aws-cdk/aws-eks-v2-alpha/lib/cluster.ts

```diff
@@ -1671,18 +1673,20 @@ export class Cluster extends ClusterBase {
 
     this.authenticationMode = props.authenticationMode;
 
-    const resource = this._clusterResource = new ClusterResource(this, 'Resource', {
+    const resource = this._clusterResource = new CfnCluster(this, 'Resource', {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The Kubernetes API server for the Amazon EKS cluster is exposed if `endpointAccess` is set to `EndpointAccess.PUBLIC` in the `Cluster` resource or `endpointPublicAccess` is `true` in the `CfnCluster` resource. This configuration allows unauthorized internet access, increasing security risks such as data breaches and malicious activities. To secure the cluster, set `endpointAccess` to `EndpointAccess.PRIVATE` in the `Cluster` resource and `endpointPublicAccess` to `false` in the `CfnCluster` resource. Implement network access controls to restrict access to trusted sources within the AWS VPC. For more information refer to the documentation https://docs.aws.amazon.com/cdk/api/v1/docs/@aws-cdk_aws-eks.EndpointAccess.html.